### ISSUE TITLE
[Blue Burst] Poison Lillies are always dropping special items

### ISSUE THREAD

**Issue by nolrinale** [fuzziqersoftware/newserv#409]
**Describe the bug**
Playing Solo Hard EP1 Caves the Poison Lillies are dropping a red box every time you kill them, it looks like the server thinks they are Nar Lillies from ultimate 

**To reproduce**
Fill in steps to reproduce the behavior, such as:
1. Create EP1 Solo/Online Hard Mode Room and go to Caves
2. Kill Lillies
3. Get Red Box!

**Game version(s) (choose one or more of the following):**
BB

**Server log output**
```
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: 0000 | 14 00 60 00 00 00 00 00 0A 03 35 11 35 01 8D 01 |   `       5 5
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: 0010 | 00 0A 00 20                                     |
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [Commands] Received from C-F (VIANNE) (version=BB command=0060 flag=00000000)
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: 0000 | 14 00 60 00 00 00 00 00 C8 03 35 11 35 01 00 00 |   `       5 5
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: 0010 | 01 00 00 00                                     |
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [C-F] EXP requested for E-135 => [Map::Enemy E-135 source 49 NAR_LILY RARE floor=03 flags=00 last_hit_by_client_id=0]
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [Commands] Sending to C-F (VIANNE) (version=BB command=0060 flag=00000000)
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: 0000 | 10 00 60 00 00 00 00 00 BF 02 00 00 03 01 00 00 |   `
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [Commands] Received from C-F (VIANNE) (version=BB command=0062 flag=00000000)
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: 0000 | 20 00 62 00 00 00 00 00 60 06 00 00 03 0D 35 01 |   b     `     5
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: 0010 | A7 9D 84 44 79 59 36 44 3C 00 00 00 03 00 00 00 |    DyY6D<
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [C-F] Drop check for E-135 NAR_LILY
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: W 359425 2024-02-27 17:48:41 - [C-F] rt_index 0D from command does not match entity's expected index 0E
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [Game:1E] Creating item from enemy 0135 (area 03)
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [ItemCreator:BB_V4/Ep1/Nml/H/0] Enemy type: E
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [ItemCreator:BB_V4/Ep1/Nml/H/0] Drop chosen (61 < 100)
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [ItemCreator:BB_V4/Ep1/Nml/H/0] Enemy spec E0000000 produced item 000406
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [Game:1E] Entity 0135 (area 03) created item ???? VJAYA
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [Game:1E] Creating item CC00008D at 03:1060.93,729.398 for all clients
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [Lobby:0000001E:FloorItems:03] Added floor item CC00008D at 1060.93, 729.398 with drop number 34 visible to 00F
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [Lobby:0000001E:FloorItems:03] Evicted 0 items
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [Commands] Sending to C-F (VIANNE) (version=BB command=0060 flag=00000000)
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: 0000 | 34 00 60 00 00 00 00 00 5F 0B 00 00 03 01 35 01 | 4 `     _     5
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: 0010 | A7 9D 84 44 79 59 36 44 00 00 00 00 00 04 06 00 |    DyY6D
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: 0020 | 80 00 00 00 00 00 00 00 8D 00 00 CC 00 00 00 00 |
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: 0030 | 00 00 00 00                                     |
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [Commands] Received from C-F (VIANNE) (version=BB command=0060 flag=00000000)
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: 0000 | 10 00 60 00 00 00 00 00 76 02 35 11 03 00 08 00 |   `     v 5
Feb 27 17:48:41 kiyomizu.moe newserv[359425]: I 359425 2024-02-27 17:48:41 - [Commands] Received from C-F (VIANNE) (version=BB command=0060 flag=00000000)


```

**Additional context**
I checked the rest of the caves monsters but they seem fine only happens with the lillies
the full room log
[lillyFullLog.txt](https://github.com/fuzziqersoftware/newserv/files/14424061/lillyFullLog.txt)

**Comment by Elixir70** [comment]
Nar Lily is the rare version of N-VH Poison Lily. Of the enemies with rare monster variants, Poison Lily is the only one that doesn't support placing a fixed one on the map. The others all require setting a field in their init data.

This bug is probably because the server is treating them as 'fixed rare monsters' here if a param is set in their init data: https://github.com/fuzziqersoftware/newserv/blob/2f5d547c19769dd284e9a4f91b6da66d7037552c/src/Map.cc#L907

This is also a bug in pub teth, and it was incorrectly documented in some old/deleted Qedit forum posts iirc. And there are many Lillies in free field and even V1/V2 quests with that param set (presumably because they were originally a different enemy like Pal or Guil Shark based on their other params).

**Comment by fuzziqersoftware** [comment]
Thanks for the context - I went through the rare enemies' constructors on V3 and BB a few months ago, but I must have missed this one on BB. I've fixed this in c8041558.
